File: 25 Nov Xor of all Subarray/code.py
class Solution:
    def subarrayXor(self, arr):
        # code here 
        n=len(arr)
        res=0
        for i,val in enumerate(arr):
            contri=(i+1)*(n-i)
            if(contri%2):res^=val
        return res
        

File: 5 Jan 2026 Max Sum Subarray of size K/code.cpp
class Solution {
  public:
    int maxSubarraySum(vector<int>& arr, int k) {
        // code here
        int sm=0,res=0;
        for(int i=0,j=0;i<arr.size();i++){
            if((i-j+1)>k)sm-=arr[j++];
            sm+=arr[i];
            res=max(res,sm);
        }
        return res;
    }
};